packages/core/src/combinator/multicast.js

```diff
@@ -0,0 +1,98 @@
+import { append, remove, findIndex } from '@most/prelude'
+import { tryEvent, tryEnd } from '../source/tryEvent'
+
+const dispose = disposable => disposable.dispose()
+
+const emptyDisposable = {
+  dispose () {}
+}
+
+class MulticastDisposable {
+  constructor (source, sink) {
+    this.source = source
+    this.sink = sink
+    this.disposed = false
+  }
+
+  dispose () {
+    if (this.disposed) {
+      return
+    }
+    this.disposed = true
+    const remaining = this.source.remove(this.sink)
+    return remaining === 0 && this.source._dispose()
+  }
+}
```

Comment:
MulticastDisposable includes some of the functionality of [`disposeOnce`](https://github.com/mostjs/core/blob/master/packages/disposable/src/disposeOnce.js).  We could potentially refactor it to just deal with multicast dispose concerns and then wrap it with `disposeOnce`.  E.g. it could be something like:

```js
class MulticastDisposable {
  constructor (source, sink) {
    this.source = source
    this.sink = sink
  }

  dispose () {
    if(this.source.remove(this.sink) === 0) {
      this.source._dispose()
    }
  }
}
```

And then below:

```js
return disposeOnce(new MulticastDisposable(this, sink))
```

I don't think it's strictly necessary for this PR, but could be a nice simplification.  What do you think?